### ISSUE TITLE
500 Error when login with steam

### ISSUE THREAD

**Issue by MadeBaruna** [odota/core#1206]
I got 500 Error tidehunter when trying to log in with my steam account. my profile https://www.opendota.com/players/78896982

**Comment by howardchung** [comment]
Could be a temporary issue with Steam authentication.  Does it persist (keep happening on retry?)

**Comment by MadeBaruna** [comment]
I cannot login since yasp changed to opendota

**Comment by howardchung** [comment]
I just tried logging in and it worked.  Can you retry?

**Comment by MadeBaruna** [comment]
Still cannot login.

**Comment by howardchung** [comment]
Can you try on my test environment?

http://hchung.westus.cloudapp.azure.com:5000/

**Comment by MadeBaruna** [comment]
I got error returned from steam API

**Comment by howardchung** [comment]
http://api.steampowered.com/ISteamUser/GetPlayerSummaries/v0002/?key=X&steamids=76561198039162710

I checked the player endpoint and it appears to be fine.

**Comment by MadeBaruna** [comment]
I don't know why then :/

**Comment by howardchung** [comment]
Can anyone else reproduce this?

**Comment by howardchung** [comment]
@MadeBaruna is this still happening?

**Comment by MadeBaruna** [comment]
yep
http://puu.sh/riMjv/eb3003150d.jpg

**Comment by howardchung** [comment]
I'll do some more investigation (probably tomorrow).

Out of curiosity, I notice your personaname is "Skill : 404 Not Found".  This is a long shot, but possibly some code in the chain is parsing that as an error?  Could you try changing your name temporarily and then trying to log in?

**Comment by MadeBaruna** [comment]
Yep i can login now, maybe because the : character?

**Comment by howardchung** [comment]
If you have time, perhaps try just "404 Not Found" to see if it's the 404 bit or the `:` bit.  This is a really weird bug though

**Comment by MadeBaruna** [comment]
You are right. 404 Not Found caused the error

**Comment by howardchung** [comment]
Yep, I confirmed.  I found this code in the steam-web dependency that passport-steam relies on:
```
      //TODO Use response code from headers instead
      if (resData.indexOf('404 Not Found') != -1) {
        callback('error returned from steam API');
        return;
      }
```

I checked the source and it appears to have been fixed, but the package containing this fix is 0.3.0 and passport-steam has not been updated to use it (0.2.4).  I submitted a pull request to update the package.
https://github.com/liamcurry/passport-steam/pull/45

**Comment by howardchung** [comment]
Another approach is to use passport-openid directly, as described:
https://github.com/liamcurry/passport-steam/issues/17#issuecomment-143535958

This also has the advantage of not failing when the Steam API is down.  However, it means we won't be able to update personaname etc. when the user logs in.
